aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -298,6 +298,59 @@ Tensor& sigmoid_out(Tensor& result, const Tensor& self) { return unary_op_impl_o
 Tensor sigmoid(const Tensor& self) { return unary_op_impl(self, at::sigmoid_out);  }
 Tensor& sigmoid_(Tensor& self) { return unary_op_impl_(self, at::sigmoid_out);  }
 
+Tensor& heaviside_tensor_out(Tensor& result, const Tensor& self, const Tensor& val) {
+  TORCH_CHECK(!self.is_complex() && !result.is_complex() && !val.is_complex(),
+    "heaviside is not implemented for complex tensors.");
+  TORCH_CHECK(result.is_floating_point(),
+    "heaviside is only implemented for floating point types output.");
+
+  result = result.to(kFloat);
+  TensorIterator iter = TensorIteratorConfig()
+    .check_all_same_dtype(false)
+    .set_check_mem_overlap(true)
+    .add_output(result)
+    .add_input(self.to(kFloat))
+    .add_input(val.to(kFloat))
+    .build();
+  heaviside_tensor_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor& heaviside_scalar_out(Tensor& result, const Tensor& self, Scalar val) {
```

Comment:
Can you wrap the scalar value in a tensor and just call the `Tensor, Tensor, Tensor` variant of this function?

Reply:
Addressed!